File: frontend/src/types/employee.ts
// Соответствует dto/employee/EmployeeResponse.java и EmployeeRequest.java
export interface EmployeeResponse {
  id: number;
  fullName: string;
  position: string | null;
  department: string | null;
  internalPhone: string | null;
  email: string | null;
  isActive: boolean | null;
}

export interface EmployeeRequest {
  fullName: string;
  position: string | null;
  department: string | null;
  internalPhone: string | null;
  email: string | null;
  isActive: boolean | null;
}

// Соответствует dto/employee/DepartmentSummaryResponse.java
export interface DepartmentSummary {
  department: string;
  employeesCount: number;
}

File: backend/src/main/java/by/GroiroTechInventory/entity/Employee.java
package by.GroiroTechInventory.entity;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;

@Entity
@Table(name = "employees")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Employee {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "full_name", nullable = false, length = 150)
    private String fullName;

    @Column(length = 100)
    private String position;

    @Column(length = 100)
    private String department;

    @Column(name = "internal_phone", length = 20)
    private String internalPhone;

    @Column(length = 100)
    private String email;

    @Builder.Default
    @Column(name = "is_active")
    private Boolean isActive = true;

    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;
}
